lib/src/badge.dart

```diff
@@ -268,6 +274,80 @@ class BadgeState extends State<Badge> with SingleTickerProviderStateMixin {
     return widget.gradient == null ? _badgeView() : _badgeViewGradient();
   }
 
+  BadgePosition? getPosition(BadgePosition? position) {
+    double? top, bottom, start, end;
+    if (position != null) {
+      if (position.isCenter) {
+        return position;
+      }
+      if (position.start != null) {
+        if (!position.start!.isNegative) {
+          start = position.start!;
+        } else {
+          start = 0;
+        }
+      }
```

Comment:
It would be better if you move this check and assignment to the method, thus removing code duplication